### PR TITLE
Updated Test cases

### PR DESCRIPTION
**Pull request by raheel4839**
Hi,

Changes proposed in this PR are related to the fixes that are required to run the test cases with the changed implementation of the plugin.

There are still some test cases which I am not sure about whether its outdated assertion or implementation should be different... for those cases comment blocks are added in `cf_cli_java_plugin_test.go` .  In comment blocks you can find the assertions that were made initially and the current values against those assertions with new implementation.

Some assertion are quite obvious e.g. non empty output responses, but some of them need more attention, especially the cases where executed commands are checked.

One way is to make everything consistent with current implementation but this might not be correct in all cases. Therefore, a review is important here.

Incase, more information is required regarding any of the comments , feel free to reach out.

Thanks,
Raheel

### CODE REVIEW

**Comment by TimGerlach** [comment]
Hi @raheel4839,
we'll have a look at this PR shortly. Thanks for your patience.

File: cf_cli_java_plugin_test.go [hunk 1]
```diff
@@ -177,13 +175,31 @@ var _ = Describe("CfJavaPlugin", func() {
...
+
+					/*
```
Comment (JohnnyVim) [inline cf_cli_java_plugin_test.go:186]: This was changed in a recent commit: https://github.com/SAP/cf-cli-java-plugin/commit/036a5eacddab82bf519139d598b28875ddf73fa0#diff-5aa2577361069a3cc3969893474e77c83b3af621509137e5cba9e8b57fee82b8L192
Due to the tests not having been executed, this was not caught. The executed command below looks like what should be expected here.

File: cf_cli_java_plugin_test.go [hunk 2]
```diff
@@ -200,10 +216,28 @@ var _ = Describe("CfJavaPlugin", func() {
...
...e:]`; if [ -n \"${JMAP_COMMAND}\" ]; then true; OUTPUT=$( ${JMAP_COMMAND} -dump:format=b,file=/tmp/heapdump-abcd-123456.hprof $(pidof java) ) || STATUS_CODE=$?; if [ ! -s /tmp/heapdump-abcd-123456.hprof ]; then echo >&2 ${OUTPUT}; exit 1; fi; if [ ${STATUS_CODE:-0} -gt 0 ]; then echo >&2 ${OUTPUT}; exit ${STATUS_CODE}; fi; cat /tmp/heapdump-abcd-123456.hprof; exit 0; fi; JVMMON_COMMAND=`find -executable -name jvmmon | head -1 | tr -d [:space:]`; if [ -n \"${JVMMON_COMMAND}\" ]; then true; OUTPUT=$( ${JVMMON_COMMAND} -pid $(pidof java) -c \"dump heap\" ) || STATUS_CODE=$?; sleep 5; HEAP_DUMP_NAME=`find -name 'java_pid*.hprof' -printf '%T@ %p\\0' | sort -zk 1nr | sed -z 's/^[^ ]* //' | tr '\\0' '\\n' | head -n 1`; SIZE=-1; OLD_SIZE=$(stat -c '%s' \"${HEAP_DUMP_NAME}\"); while [ \"${SIZE}\" != \"${OLD_SIZE}\" ]; do sleep 3; SIZE=$(stat -c '%s' \"${HEAP_DUMP_NAME}\"); done; if [ ! -s \"${HEAP_DUMP_NAME}\" ]; then echo >&2 ${OUTPUT}; exit 1; fi; if [ ${STATUS_CODE:-0} -gt 0 ]; then echo >&2 ${OUTPUT}; exit ${STATUS_CODE}; fi; cat ${HEAP_DUMP_NAME}; fi; rm -f /tmp/heapdump-abcd-123456.hprof; if [ -n \"${HEAP_DUMP_NAME}\" ]; then rm -f ${HEAP_DUMP_NAME} ${HEAP_DUMP_NAME%.*}.addons; fi"}))
```
Comment (JohnnyVim) [inline cf_cli_java_plugin_test.go:227]: Same issue as above: due to recent code changes, the expected command needs to be adjusted accordingly

File: cf_cli_java_plugin_test.go [hunk 3]
```diff
@@ -280,13 +334,32 @@ var _ = Describe("CfJavaPlugin", func() {
...
...executable -name jmap | head -1 | tr -d [:space:]`; JVMMON_COMMAND=`find -executable -name jvmmon | head -1 | tr -d [:space:]`; if [ -n \"${JMAP_COMMAND}\" ]; then true; OUTPUT=$( ${JMAP_COMMAND} -dump:format=b,file=/tmp/my_app-heapdump-abcd-123456.hprof $(pidof java) ) || STATUS_CODE=$?; if [ ! -s /tmp/my_app-heapdump-abcd-123456.hprof ]; then echo >&2 ${OUTPUT}; exit 1; fi; if [ ${STATUS_CODE:-0} -gt 0 ]; then echo >&2 ${OUTPUT}; exit ${STATUS_CODE}; fi; elif [ -n \"${JVMMON_COMMAND}\" ]; then true; echo -e 'change command line flag flags=-XX:HeapDumpOnDemandPath=/tmp\ndump heap' > setHeapDumpOnDemandPath.sh; OUTPUT=$( ${JVMMON_COMMAND} -pid $(pidof java) -cmd \"setHeapDumpOnDemandPath.sh\" ) || STATUS_CODE=$?; sleep 5; HEAP_DUMP_NAME=`find /tmp -name 'java_pid*.hprof' -printf '%T@ %p\\0' | sort -zk 1nr | sed -z 's/^[^ ]* //' | tr '\\0' '\\n' | head -n 1`; SIZE=-1; OLD_SIZE=$(stat -c '%s' \"${HEAP_DUMP_NAME}\"); while [ ${SIZE} != ${OLD_SIZE} ]; do OLD_SIZE=${SIZE}; sleep 3; SIZE=$(stat -c '%s' \"${HEAP_DUMP_NAME}\"); done; if [ ! -s \"${HEAP_DUMP_NAME}\" ]; then echo >&2 ${OUTPUT}; exit 1; fi; if [ ${STATUS_CODE:-0} -gt 0 ]; then echo >&2 ${OUTPUT}; exit ${STATUS_CODE}; fi; fi",
```
Comment (JohnnyVim) [inline cf_cli_java_plugin_test.go:358]: Needs to be adjusted due to recent code changes.

File: cf_cli_java_plugin_test.go [hunk 4]
```diff
@@ -301,14 +374,35 @@ var _ = Describe("CfJavaPlugin", func() {
...
-						"JVMMON_COMMAND=`find -executable -name jvmmon | head -1 | tr -d [:space:]`; if [ -n \"${JVMMON_COMMAND}\" ]; then true; OUTPUT=$( ${JVMMON_COMMAND} -pid $(pidof java) -c \"dump heap\" ) || STATUS_CODE=$?; sleep 5; HEAP_DUMP_NAME=`find -name 'java_pid*.hprof' -printf '%T@ %p\\0' | sort -zk 1nr | sed -z 's/^[^ ]* //' | tr '\\0' '\\n' | head -n 1`; SIZE=-1; OLD_SIZE=$(stat -c '%s' \"${HEAP_DUMP_NAME}\"); while [ \"${SIZE}\" != \"${OLD_SIZE}\" ]; do sleep 3; SIZE=$(stat -c '%s' \"${HEAP_DUMP_NAME}\"); done; if [ ! -s \"${HEAP_DUMP_NAME}\" ]; then echo >&2 ${OUTPUT}; exit 1; fi; if [ ${STATUS_CODE:-0} -gt 0 ]; then echo >&2 ${OUTPUT}; exit ${STATUS_CODE}; fi; cat ${HEAP_DUMP_NAME}; fi'"
-
-					Expect(output).To(Equal(expectedOutput))
+					/*
+						-----------------------------------------------------------------------------------------------------------------------
+																		Test Case Issues
+						-----------------------------------------------------------------------------------------------------------------------
+							expectedOutput := "cf ssh my_app --app-instance-index 4 --command '" + JavaDetectionCommand + "; " +
```
Comment (JohnnyVim) [inline cf_cli_java_plugin_test.go:381]: Needs to be adjusted due to recent code changes

**Review by JohnnyVim — CHANGES_REQUESTED** [review]

File: cf_cli_java_plugin_test.go [hunk 5]
```diff
@@ -219,11 +253,12 @@ var _ = Describe("CfJavaPlugin", func() {
 					})
 
 					Expect(output).To(BeEmpty())
-					Expect(err).To(ContainSubstring("the container path specified doesn't exist or have no read and write access, please check and try again later"))
+					Expect(err.Error()).To(ContainSubstring("the container path specified doesn't exist or have no read and write access, please check and try again later"))
 					Expect(cliOutput).To(ContainSubstring("the container path specified doesn't exist or have no read and write access, please check and try again later"))
 
 					Expect(commandExecutor.ExecuteCallCount()).To(Equal(0))
-					Expect(commandExecutor.ExecuteArgsForCall(0)).To(Equal(0))
+					// Expect(commandExecutor.ExecuteArgsForCall(0)).To(Equal(0))
```
[the remark this answers is not in the document]
Comment (raheel4839) [reply cf_cli_java_plugin_test.go]: suggested change has been incorporated in code.

File: cf_cli_java_plugin_test.go [hunk 6]
```diff
@@ -62,8 +60,8 @@ var _ = Describe("CfJavaPlugin", func() {
 			subject = &JavaPlugin{}
 			commandExecutor = new(FakeCommandExecutor)
 			uuidGenerator = new(FakeUUIDGenerator)
-			pluginUtil = FakeCfJavaPluginUtil{SshEnabled: true, Jmap_jvmmon_present: true, Container_path_valid: true, Fspath: "/tmp", LocalPathValid: true}
 			uuidGenerator.GenerateReturns("abcd-123456")
+			pluginUtil = FakeCfJavaPluginUtil{SshEnabled: true, Jmap_jvmmon_present: true, Container_path_valid: true, Fspath: "/tmp", LocalPathValid: true, UUID: "abcd-123456", OutputFileName: "java_pid0_0.hprof"}
```
[the remark this answers is not in the document]
Comment (raheel4839) [reply cf_cli_java_plugin_test.go]: alternate approach has been incorporated in code.

File: cf_cli_java_plugin_test.go [hunk 7]
```diff
@@ -239,11 +274,29 @@ var _ = Describe("CfJavaPlugin", func() {
...
+					// Expect(err.Error()).To(ContainSubstring("Error creating local file at : /not/valid/path. Please check that you are allowed to create files at the given local path."))
+					// Expect(cliOutput).To(ContainSubstring("Error creating local file at : /not/valid/path. Please check that you are allowed to create files at the given local path."))
+
+					// change in above 2 assertions.
+					// heap dump file created under this scenario are not deleted from the container.
+					Expect(err.Error()).To(ContainSubstring("Error occured during create desination file: /not/valid/path/my_app-heapdump-" + pluginUtil.UUID + ".hprof, please check you are allowed to create file in the path."))
+					Expect(cliOutput).To(ContainSubstring("Successfully created heap dump in application container at: " + pluginUtil.Fspath + "/" + pluginUtil.OutputFileName + "|FAILED|Error occured during create desination file: /not/valid/path/my_app-heapdump-" + pluginUtil.UUID + ".hprof, please check you are allowed to create file in the path.|"))
```
[the remark this answers is not in the document]
Comment (raheel4839) [reply cf_cli_java_plugin_test.go:244]: you are right this is the prod code behavior at the moment and out of the scope of this PR. So, as discussed relevant changes are made in the code.

File: cf_cli_java_plugin_test.go (hunk 1 above)
Comment (raheel4839) [reply cf_cli_java_plugin_test.go]: suggested change has been incorporated in code.

File: cf_cli_java_plugin_test.go (hunk 3 above)
Comment (raheel4839) [reply cf_cli_java_plugin_test.go]: suggested change has been incorporated in code.

File: cf_cli_java_plugin_test.go (hunk 2 above)
Comment (raheel4839) [reply cf_cli_java_plugin_test.go]: suggested change has been incorporated in code.

**Review by raheel4839 — COMMENTED** [review]
changes reviewed and suggested change has been incorporated in code.

**Review by JohnnyVim — APPROVED** [review]
Very nice, LGTM!

**Pull request merged**